Modules/_ctypes/_ctypes.c

```diff
@@ -3774,19 +3794,52 @@ PyCFuncPtr_FromDll(PyTypeObject *type, PyObject *args, PyObject *kwds)
         return NULL;
     }
 #else
+    #ifdef __CYGWIN__
+        //dlerror() isn't very helpful on cygwin */
+    #else
+        #define USE_DLERROR
+        /* dlerror() always returns the latest error.
+         *
+         * Clear the previous value before calling dlsym(),
+         * to ensure we can tell if our call resulted in an error.
+         */
+        (void)dlerror();
+    #endif
     address = (PPROC)dlsym(handle, name);
-    if (!address) {
-#ifdef __CYGWIN__
-/* dlerror() isn't very helpful on cygwin */
+
+    if (address) {
+        goto dlsym_ok;
+    }
+    else {
```

Comment:
Hm, am I missing something?
All the non-Windows error handling is in one block, so you shouldn't need the `goto` and label.
```suggestion
    if (!address) {
```

Reply:
I think I was the one who missed something! Fixed.